src/repo/kv.rs

```diff
@@ -101,234 +95,301 @@ impl DataStore for KvDataStore {
     }
 }
 
+// in the transactional parts of the [`Infallible`] is used to signal there is no additional
+// custom error, not that the transaction was infallible in itself.
+
 #[async_trait]
 impl PinStore for KvDataStore {
-    async fn is_pinned(&self, block: &Cid) -> Result<bool, Error> {
-        is_pinned(self, block)
+    async fn is_pinned(&self, cid: &Cid) -> Result<bool, Error> {
```

Comment:
however, this looking like it can be just an atomic read makes me think it must be the wrong abstraction.. but if I recall correctly, this is called when removing a block through `/api/v0/block/rm` so ... I guess it needs to stay for now.